tests/plugins/test_training_type_plugin_teardown.py

```diff
@@ -0,0 +1,56 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import torch
+
+from pytorch_lightning import Trainer
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.runif import RunIf
+from tests.helpers.utils import pl_multi_process_test
+
+
+class BoringModelGPUTearDown(BoringModel):
+
+    def on_train_start(self) -> None:
+        assert self.device == torch.device("cuda:0")
+        self.start_cuda_memory = torch.cuda.memory_allocated()
+
+    def on_fit_end(self) -> None:
+        assert self.device == torch.device("cpu")
+        cuda_memory = torch.cuda.memory_allocated()
+        assert cuda_memory < self.start_cuda_memory
+
+
+@RunIf(skip_windows=True, min_gpus=1)
+def test_single_gpu():
+    """Tests if teardown correctly for single GPU plugin."""
+    trainer = Trainer(gpus=1, fast_dev_run=True)
+    model = BoringModelGPUTearDown()
+    trainer.fit(model)
+
+
+class BoringModelTPUTearDown(BoringModel):
+
+    def on_fit_end(self) -> None:
+        assert "PT_XLA_DEBUG" not in os.environ
```

Comment:
this is called before teardown: https://github.com/PyTorchLightning/pytorch-lightning/blob/a8d9b5f783528c2b18407d42298e10d2c7b18c61/pytorch_lightning/trainer/trainer.py#L776-L784

we could assert that PT_XLA_DEBUG is in the environ during training, and that it's not after `trainer.fit()` completes in L56

Reply:
Oh, the L784, is for model teardown, accelerator teardown is actually here: https://github.com/PyTorchLightning/pytorch-lightning/blob/a8d9b5f783528c2b18407d42298e10d2c7b18c61/pytorch_lightning/trainer/trainer.py#L804

which is in `_post_dispatch`

It is fine also for your proposal! Added `on_fit_end` as would like to test immediately after accelerator teardown.